### DISCUSSION TITLE
How to save while downloading?

### DISCUSSION THREAD

**Discussion by failable** [spider-rs/spider#177] [Q&A] [answered]
Looking into the example `download.rs`, it seems that one can save the HTMLs unless all the pages are scraped? Can one save downloaded HTMLs while downloading new pages? Thanks. (Pretty new to Rust, maybe dumb question.)

**Comment by j-mendez** [comment, marked as answer]
You can use subscriptions and store the file.

**Reply by failable** [reply]
Thanks!

**Reply by failable** [reply]
Hello, I have one more question. How to set `capacity` in `subscribe` to avoid missing pages or how should I handle the message properly in limited channel size? Should I set it to sufficient large size? I read the documentation of `broadcast` module of `tokio` but I did not find a clear way to handle it in this scrape case.

**Reply by j-mendez** [reply]
Use website.subscribe_guard, this will process all until complete or await the task handle.
